### PR TITLE
Migrate dissolution tank & digester to processingLogic

### PR DESCRIPTION
**Pull request by GDCloudstrike**
Migrates both multis to processingLogic so features like batch mode & void protection are available

### CODE REVIEW

**Comment by Elisis** [comment]
Does batch mode work as expected with the dissolution tank's input ratio logic?

**Comment by GDCloudstrike** [comment]
> Does batch mode work as expected with the dissolution tank's input ratio logic?

yes, as it checks for the ratio between the 2 fluids and not their absolute values
the only change in behaviour is that instead of voiding the recipe if the ratios aren't correct, the machine now just doesnt start

image

File: src/main/java/com/elisis/gtnhlanth/common/tileentity/DissolutionTank.java [hunk 1]
```diff
@@ -96,36 +104,39 @@ public RecipeMap<?> getRecipeMap() {
...
-                .findRecipe(getBaseMetaTileEntity(), false, tVoltage, tFluidInputArray, tItems);
+    protected ProcessingLogic createProcessingLogic() {
+        return new ProcessingLogic() {
 
-        if (tRecipe == null || !tRecipe.isRecipeInputEqual(true, tFluidInputArray, tItems)) return false;
-        // GT_Log.out.print("Recipe not null\n");
+            @NotNull
+            @Override
+            protected CheckRecipeResult validateRecipe(@Nonnull GT_Recipe recipe) {
+                List<FluidStack> fluidInputs = DissolutionTank.this.getStoredFluids();
```
Comment (HoleFish) [inline src/main/java/com/elisis/gtnhlanth/common/tileentity/DissolutionTank.java]: `inputFluids` is set up in `ProcessingLogic`.

**Comment by AbdielKavash** [comment]
> the only change in behaviour is that instead of voiding the recipe if the ratios aren't correct, the machine now just doesnt start

Doesn't that essentially invalidate the automation challenge with it? As long as the player inputs fluid slow enough, one bucket at a time, will the DT simply start when the correct ratio is reached?

**Comment by GDCloudstrike** [comment]
> > the only change in behaviour is that instead of voiding the recipe if the ratios aren't correct, the machine now just doesnt start
> 
> Doesn't that essentially invalidate the automation challenge with it? As long as the player inputs fluid slow enough, one bucket at a time, will the DT simply start when the correct ratio is reached?

WIth the current implementation that is the case, yes. If this is not wanted I can make it void and stop again.

**Comment by Elisis** [comment]
> > > the only change in behaviour is that instead of voiding the recipe if the ratios aren't correct, the machine now just doesnt start
> > 
> > 
> > Doesn't that essentially invalidate the automation challenge with it? As long as the player inputs fluid slow enough, one bucket at a time, will the DT simply start when the correct ratio is reached?
> 
> WIth the current implementation that is the case, yes. If this is not wanted I can make it void and stop again.

Yeah if you could do that, that'd be more in line with how the multi's meant to behave.

**Review by Elisis — APPROVED** [review]

**Pull request merged**